TestTriggers flake

https://ci.openshift.redhat.com/jenkins/job/merge_pull_requests_origin/3645/console
--- FAIL: TestTriggers_imageChange-2 (5.89s)
	deploy_trigger_test.go:165: Waiting for image stream mapping to be reflected in the IS status...
	deploy_trigger_test.go:175: Still waiting for latest tag status on ImageStream test-image-stream
	deploy_trigger_test.go:172: ImageStream test-image-stream now has Status with tags: map[string]api.TagEventList{"latest":api.TagEventList{Items:[]api.TagEvent{api.TagEvent{Created:util.Time{Time:time.Time{sec:63580568103, nsec:0, loc:(*time.Location)(0x42caa60)}}, DockerImageReference:"registry:8080/openshift/test-image@sha256:00000000000000000000000000000001", Image:"sha256:00000000000000000000000000000001"}}}}
	deploy_trigger_test.go:188: Waiting for a new deployment config in response to ImageStream update
	deploy_trigger_test.go:201: expected config version 1, got 0

--- a/test/integration/deploy_trigger_test.go
+++ b/test/integration/deploy_trigger_test.go
@@ -145,17 +145,19 @@ func TestTriggers_imageChange(t *testing.T) {
 	}
 	defer imageWatch.Stop()
 
+	updatedImage := "sha256:00000000000000000000000000000001"
+	updatedPullSpec := fmt.Sprintf("registry:8080/openshift/test-image@%s", updatedImage)
 	// Make a function which can create a new tag event for the image stream and
 	// then wait for the stream status to be asynchronously updated.
-	createTagEvent := func(image string) {
+	createTagEvent := func() {
 		mapping := &imageapi.ImageStreamMapping{
 			ObjectMeta: kapi.ObjectMeta{Name: imageStream.Name},
 			Tag:        "latest",
 			Image: imageapi.Image{
 				ObjectMeta: kapi.ObjectMeta{
-					Name: image,
+					Name: updatedImage,
 				},
-				DockerImageReference: fmt.Sprintf("registry:8080/openshift/test-image@%s", image),
+				DockerImageReference: updatedPullSpec,
 			},
 		}
 		if err := openshiftProjectAdminClient.ImageStreamMappings(testutil.Namespace()).Create(mapping); err != nil {
@@ -182,7 +184,7 @@ func TestTriggers_imageChange(t *testing.T) {
 		t.Fatalf("Couldn't create DeploymentConfig: %v", err)
 	}
 
-	createTagEvent("sha256:00000000000000000000000000000001")
+	createTagEvent()
 
 	var newConfig *deployapi.DeploymentConfig
 	t.Log("Waiting for a new deployment config in response to ImageStream update")
@@ -192,13 +194,17 @@ waitForNewConfig:
 		case event := <-configWatch.ResultChan():
 			if event.Type == watchapi.Modified {
 				newConfig = event.Object.(*deployapi.DeploymentConfig)
-				break waitForNewConfig
+				// Multiple updates to the config can be expected (e.g. status
+				// updates), so wait for a significant update (e.g. version).
+				if newConfig.LatestVersion > 0 {
+					if e, a := updatedPullSpec, newConfig.Template.ControllerTemplate.Template.Spec.Containers[0].Image; e != a {
+						t.Fatalf("unexpected image for pod template container 0; expected %q, got %q", e, a)
+					}
+					break waitForNewConfig
+				}
+				t.Log("Still waiting for a new deployment config in response to ImageStream update")
 			}
 		}
-	}
-
-	if e, a := 1, newConfig.LatestVersion; e != a {
-		t.Fatalf("expected config version %d, got %d", e, a)
 	}
 }
 

